fluss-common/src/main/java/org/apache/fluss/row/arrow/writers/ArrowArrayWriter.java

```diff
@@ -41,7 +41,11 @@ public void doWrite(int rowIndex, DataGetters row, int ordinal, boolean handleSa
         listVector.startNewValue(rowIndex);
         for (int arrIndex = 0; arrIndex < array.size(); arrIndex++) {
             int fieldIndex = offset + arrIndex;
-            elementWriter.write(fieldIndex, array, arrIndex, handleSafe);
+            // Always use safe writes for array elements because the element index (offset +
```

Comment:
We should mention in the comments on the class that the handleSafe field is ignored when writing.